plonky2/src/hash/hashing.rs

```diff
@@ -136,9 +114,6 @@ pub fn hash_n_to_m<F: RichField, P: PlonkyPermutation<F>>(
     }
 }
```

Comment:
This function returns `[0,0,0,0]` when `inputs` is empty. In particular `permute` isn't called on empty inputs. I'm not sure if it's a big issue, but it's the reason some tests are failing. Since the public inputs are hashed without padding now, any circuit without public inputs doesn't use a `PoseidonGate` anymore which makes some buggy tests fail.

Reply:
Oh thanks for investigating! I'll wait for #469 to land.

Reply:
#469 has shipped 😊